augur/curate/normalize_strings.py

```diff
@@ -0,0 +1,50 @@
+"""
+Normalize strings to a Unicode normalization form.
+
+Strings need to be normalized for predictable string comparisons, especially
+in cases where strings contain diacritics (see https://unicode.org/faq/normalization.html).
+"""
+import unicodedata
+
+from augur.utils import first_line
+
+
+def register_parser(parent_subparsers):
+    parser = parent_subparsers.add_parser("normalize-strings",
+        help=first_line(__doc__))
+
+    optional = parser.add_argument_group(title="OPTIONAL")
+    optional.add_argument("--form", default="NFC", choices=["NFC", "NFKC", "NFD", "NFKD"],
```

Comment:
I also saw this in the [unicode normalization FAQ](https://unicode.org/faq/normalization.html):

> The choice of which to use depends on the particular program or system. NFC is the best form for general text, since it is more compatible with strings converted from legacy encodings. NFKC is the preferred form for identifiers, especially where there are security concerns (see [UTR #36](https://www.unicode.org/reports/tr36/)). NFD and NFKD are most useful for internal processing.

I hadn't thought about overriding the form on a field-by-field basis. That would be a helpful feature to add in the future!

Reply:
Ah, yeah, that quoted bit is good. NFKC is what nextstrain.org uses for group names (identifiers) exactly so that "ﬂu" and "flu" can't both exist; it'd be a security concern if they could.

Reply:
Created an [issue](https://github.com/nextstrain/augur/issues/1082) to track this for future work.